src/libraries/Microsoft.PowerFx.Core/App/ErrorContainers/ErrorHelpers.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.App.ErrorContainers
+{
+    internal static class ErrorHelpers
+    {
+        public static void TypeMismatchError(this IErrorContainer errorContainer, TexlNode node, DType expected, DType actual)
+        {
+            var expectedTypeDisambiguation = expected.IsOptionSet && expected.OptionSetInfo != null ? $"({expected.OptionSetInfo.EntityName})" : string.Empty;
+            var actualTypeDisambiguation = actual.IsOptionSet && actual.OptionSetInfo != null ? $"({actual.OptionSetInfo.EntityName})" : string.Empty;
```

Comment:
>$"({ [](http://example.com/codeflow?start=96&length=4)

Consider adding a space here (so that the message is something like `Invalid argument type (Enum (Color)). Expecting a Enum (ErrorKind) value instead.`